### ISSUE TITLE
Label parameter will crash program

### ISSUE THREAD

**Issue by AleffCorrea** [BlockoS/Etripator#56]
When running Etripator on a MinGW-32 + gcc OR MSVC++17 build under Windows, with Jansson 2.12, the program will work normally with or without a configuration file, but when a label .json file is included in the arguments, eg:

etripator.exe -l labels.json config.json "PC Engine ROM file.pce"

the program will fire a SIGSEGV at the second iteration of etripator.c:169 (the first load of the labels.json goes fine), but then the program continues the for loop and tries to read cmdOptions.labelsIn[1] which is uninitialized memory.

I tried to take a peek at the code with gdb but I really couldn't get where the issue was. I'm not sure if this is a platform specific thing.

**Comment by BlockoS** [comment]
Ok I think I found it... It's a pretty stupid one :|

Here's a quick fix:
In options.c line 88, replace
```C
memset(tmp+top, 0, capacity-count);
``` 
with 
```C
memset(tmp+top, 0, (capacity-count) * sizeof(char*));
```

**Comment by AleffCorrea** [comment]
Great! However the program hangs in another part of the code now :/

```
Program received signal SIGSEGV, Segmentation fault.
0x746353c7 in WideCharToMultiByte () from C:\WINDOWS\SysWOW64\KernelBase.dll
(gdb) backtrace
#0  0x746353c7 in WideCharToMultiByte () from C:\WINDOWS\SysWOW64\KernelBase.dll
#1  0x74634fc6 in WideCharToMultiByte () from C:\WINDOWS\SysWOW64\KernelBase.dll
#2  0x776b32f6 in wcstombs_l () from C:\WINDOWS\SysWOW64\msvcrt.dll
#3  0x776b3565 in wcstombs () from C:\WINDOWS\SysWOW64\msvcrt.dll
#4  0x0040dc21 in __mingw_basename ()
#5  0x004014a8 in PrintMsg (type=MSG_TYPE_INFO,
    file=0x4121bc <__register_frame_info+4268476> "decode.c",
    line=166, function=0x41236c <__FUNCTION__.3776> "extractLabels",
    format=0x41220f <__register_frame_info+4268559> "%s:") at message.c:69
#6  0x00401fe4 in extractLabels (decoder=0x63fec4) at decode.c:166
#7  0x00407672 in main (argc=5, argv=0x31708) at etripator.c:257
(gdb)
```

**Comment by AleffCorrea** [comment]
Inconvenience? I felt I was the one being inconvenient :p 

Everything is working fine now, thanks for taking your time to fix it (and thanks for making this disassembler! This is incredibly useful). :)

**Comment by BlockoS** [comment]
The fixes were pushed to branch v0.7
